File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int maxW = 0;
        int maxH = 0;
        for (int i = 0; i<sizes.length; i++) {
            int w = Math.max(sizes[i][0], sizes[i][1]);
            int h = Math.min(sizes[i][0], sizes[i][1]);
            
            maxW = Math.max(w, maxW);
            maxH = Math.max(h, maxH);
            
        }
        return maxW * maxH;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0
    maxW = 0
    maxH = 0
    
    for i in range(len(sizes)):
        w = max(sizes[i][0], sizes[i][1])
        h = min(sizes[i][0], sizes[i][1])
        
        maxW = max(w, maxW)
        maxH = max(h, maxH)
        
    answer = maxW * maxH
    return answer